tests/Unit/Evolution/Systems/ScalarWave/Test_Characteristics.cpp

```diff
@@ -0,0 +1,388 @@
+  pypp::check_with_random_values<1>(speed_with_index<3, Dim>, "Characteristics",
+                                    "char_speed_uminus", {{{-10.0, 10.0}}},
+                                    used_for_size);
+  pypp::check_with_random_values<1>(speed_with_index<2, Dim>, "Characteristics",
+                                    "char_speed_uplus", {{{-10.0, 10.0}}},
+                                    used_for_size);
+}
+
+// Test return-by-reference char speeds by comparing to analytic solution
+template <size_t Dim>
+void test_characteristic_speeds_analytic(
+    const size_t grid_size_each_dimension) noexcept {
+  // Setup mesh
+  Mesh<Dim> mesh{grid_size_each_dimension, Spectral::Basis::Legendre,
+                 Spectral::Quadrature::GaussLobatto};
+  const auto x_logical = logical_coordinates(mesh);
+  // Get ingredients
+  const size_t n_pts = x_logical.begin()->size();
+  // Outward 3-normal to the surface on which characteristic fields are needed
+  tnsr::i<DataVector, Dim, Frame::Inertial> unit_normal_one_form{
+      DataVector(n_pts, 1.)};
+  const auto norm_of_one_form = magnitude(unit_normal_one_form);
+  for (size_t i = 0; i < Dim; ++i) {
+    unit_normal_one_form.get(i) /= get(norm_of_one_form);
+  }
+
+  // Get characteristic speeds locally
+  const auto upsi_speed =
+      make_with_value<Scalar<DataVector>>(unit_normal_one_form, 0.);
+  const auto uzero_speed =
+      make_with_value<Scalar<DataVector>>(unit_normal_one_form, 0.);
+  const auto uplus_speed =
+      make_with_value<Scalar<DataVector>>(unit_normal_one_form, 1.);
+  const auto uminus_speed =
+      make_with_value<Scalar<DataVector>>(unit_normal_one_form, -1.);
+
+  // Check that locally computed fields match returned ones
+  const auto char_speeds_from_func =
+      ScalarWave::CharacteristicSpeedsCompute<Dim>::function(
+          unit_normal_one_form);
+  const auto& upsi_speed_from_func = char_speeds_from_func[0];
+  const auto& uzero_speed_from_func = char_speeds_from_func[1];
+  const auto& uplus_speed_from_func = char_speeds_from_func[2];
+  const auto& uminus_speed_from_func = char_speeds_from_func[3];
+
+  CHECK_ITERABLE_APPROX(upsi_speed.get(), upsi_speed_from_func);
+  CHECK_ITERABLE_APPROX(uzero_speed.get(), uzero_speed_from_func);
+  CHECK_ITERABLE_APPROX(uplus_speed.get(), uplus_speed_from_func);
+  CHECK_ITERABLE_APPROX(uminus_speed.get(), uminus_speed_from_func);
+}
+}  // namespace
+
+namespace {
+template <typename Tag, size_t Dim>
+typename Tag::type field_with_tag(
+    const Scalar<DataVector>& gamma_2, const Scalar<DataVector>& psi,
+    const Scalar<DataVector>& pi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_one_form) {
+  return get<Tag>(ScalarWave::CharacteristicFieldsCompute<Dim>::function(
+      gamma_2, psi, pi, phi, normal_one_form));
+}
+
+template <size_t Dim>
+void test_characteristic_fields() noexcept {
+  const DataVector used_for_size(20);
```

Comment:
Why not the usual value, 5?

Reply:
I didn't know we had a usual value, changed to `5` now :)